include/editor.hpp

```diff
@@ -0,0 +1,200 @@
+
+  private:
+    /* @brief Get the 2 bytes data.
+     *
+     * @returns 2 Bytes value read.
+     */
+    types::LE2ByteData
+        get2BytesValue(types::BinaryVector::const_iterator iterator);
+
+    /* @brief ECC check for VHDR.
+     *
+     * It calls helper function to validate the ECC data at ECC offset for VHDR.
+     *
+     * @returns true if ECC check passes.
+     *          false if ECC check fails.
+     */
+    bool eccCheckForVHDR();
+
+    /* @brief Validates VHDR.
+     *
+     * It checks for 'VHDR' record name at defined byte to confirm it is
+     * VPD's header.
+     *
+     * @throw DataException - if VHDR record name not found OR ECC check fails.
+     */
+    void validateHeader();
+
+    /* @brief ECC check for VTOC.
+     *
+     * It calls helper function to validate the ECC data at ECC offset for VTOC.
+     *
+     * @returns true if ECC check passes.
+     *          false if ECC check fails.
+     */
+    bool eccCheckForVtoc();
+
+    /* @brief validates VTOC.
+     *
+     * It checks for 'VTOC' record name at defined byte to confirm it is
+     * VPD's TOC.
+     *
+     * @throw DataException - if VTOC record name not found OR ECC check fails.
+     */
+    void validateVtoc();
+
+    /* @brief process through VTOC.
+     *
+     * This API process through the VTOC and checks every Record's ECC
+     * Also, it looks for requested RECORD to modify, once found it collects
+     * all the required fields.
+     * RecordOffset
+     * RecordLength
+     * RecordEccOffset
+     * RecordECCLength
+     *
+     * @throw DataException- if data offset,length is not valid.
+     * @throw EccException - if ECC check fails.
+     */
+    void processVtoc();
+
+    /* @brief Get the Keyword information.
+     *
+     * This API goes through the Record and finds out requested Keyword offset
+     * and length.
+     *
+     * @throw DataException - if keyword not found
+     */
+    void getTheKwdOffsetAndLength();
+
+    /* @brief It writes the new data to the VPD.
+     *
+     * This API writes the new data to the VPD on HW.
+     */
+    void writeNewDataToVpd();
+
+    /* @brief Updates the New ECC as per New Data.
+     *
+     * @throw EccException - if ECC update fails.
+     */
+    void updateRecordEcc();
+
+    /* @brief Updates the Cache for the reqeusted record-keyword.
+     *
+     * Cache will be updated only when requested explicitly.
+     */
+    void updateCache();
+
+    /* @brief Updates EI for the reqeusted record-keyword.
+     *
+     * If requested keyword is part of ExtraInterfaces, need to update
+     * it there also.
+     */
+    void updateEI(const nlohmann::json& singleEEPROM,
+                  const std::string& objPath);
+
+    /* @brief Updates CI for the reqeusted record-keyword.
+     *
+     * If requested keyword is part of commonInterfaces, need to update
+     * it there also.
+     */
+    void updateCI(const std::string& objectPath);
```

Comment:
we have to make use of updateCI, updateEI, udpdateCache in P11 new design, if we are getting rid of editor class.